docs/api.md

```diff
@@ -5,31 +5,88 @@ permalink: /docs/api/
 
 ```
 var compiledTemplate = dust.compile("Hello {name}!", "intro");
 ```
 
-If you include the "compiled" string as part of a script block of JS that you load, then the "intro" template will be defined and registered. If you want to do it immediately then do:
+If you include the "compiled" output as part of a script block of JS that you load, then the "intro" template will be defined and registered. 
 
 ```
-dust.loadSource(compiled);
+<script type="text/javascript">
+/* example compiled dust template output */
+(function() {
+  dust.register("demo", body_0);
+
+  function body_0(chk, ctx) {
+    return chk.write("Hello ").reference(ctx.get("name"), ctx, "h").write("! You have ").reference(ctx.get("count"), ctx, "h").write(" new messages.");
+  }
+  return body_0;
+})();
+</script>
+```
+
+If you want to register it directly from the ouput of the dust.compile function, use dust.loadSource:
+
+```
+var compiledTemplate = dust.compile("Hello {name}!", "intro");
+dust.loadSource(compiledTemplate);
 ```
 
 This will also define and register the template.  All of these compiled templates are saved in the dust.cache Array under the name they were registered under, in this case, "intro".
 
 ##Rendering a template
-Assuming you have the "intro" template we previously compiled and loaded, you can render it using the code below. The first argument to dust.render is the registered template name, the second argument is the JSON model, the third argument is a callback function where "out" contains the rendered output from running the template (and err has any error information).
+###Function call
+```    
+dust.render(templateName, data, callback);
+```
+###Parameters
+* templateName {String}: The registered name of the template to render.  This was the name given to the template at the compilation step.
+* data {Object}: The javascript context to be consumed by the template.
+* callback {Function}: The function that gets called when the render either succeeds or fails.
+* @return {Void}
+
+###Example call
+Assuming you have the "intro" template we previously compiled and loaded, you can render it using the code below.
 
 ```
 dust.render("intro", {name: "Fred"}, function(err, out) {
-    console.log(out);
+    if (err) {
+      console.log(err)
+    } else {
+      console.log(out);
+    }
 });
 ```
 
 ##Streaming a template
-Streaming can be used in place of rendering to get the outputs of a template asynchronously, such that you can set up parallel processes to get the outputted template data at blazing fast speeeds.  Streaming is asyncronous, so it the rendering is based on event callbacks.
+###Function call
+```    
+dust.stream(templateName, data);
+```
+###Parameters
+* templateName {String}: The registered name of the template to render.  This was the name given to the template at the compilation step.
+* data {Object}: The javascript context to be consumed by the template.
+* @return {Void}
```

Comment:
It does not return Void. If it did, you wouldn't be able to chain `on('error', ...)` to the `dust.stream` call.